### PR TITLE
Completes Issue #21 Add email Template

### PR DESCRIPTION
**Pull request by victorlian**
This pull request should finish the feature request in #21 Add email templates.
**As this feature required a fair amount of code to write (front-end, back-end and tests), and hence a large amount of existing code to understand, I propose upgrading this feature from medium to large.**  @softeng-701 

An email template can be applied, created, modified and deleted by a user. 
A user may only use templates created by him (not anyone else).
These templates will be stored in the database.

There will also be a default template, and a "none" (empty) template. These templates are added by the front-end and can be used by any user. 

**Example usage**
1. Applying a template when editing email:

2. Managing templates:

### CODE REVIEW

**Comment by pulkitkalra** [comment]
Tested and works as expected! Good work.

Just a couple of things that I noted while testing:
(1) It allows you to save a template with the same name (probably shouldn't be allowed?). E.g. you can create many templates called 'default'. 

(2) This is a small usability thing: It's a little tricky to work out which template is applied, since the name is in this drop-down here
image
I would think it'd be better to move it to the top just above the subject?

File: src/test/java/org/project36/qualopt/web/rest/EmailTemplateResourceIntTest.java [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        emailTemplate1.setId(1L);
+        EmailTemplate emailTemplate2 = new EmailTemplate();
+        emailTemplate2.setId(emailTemplate1.getId());
+        assertThat(emailTemplate1).isEqualTo(emailTemplate2);
+        emailTemplate2.setId(2L);
+        assertThat(emailTemplate1).isNotEqualTo(emailTemplate2);
+        emailTemplate1.setId(null);
+        assertThat(emailTemplate1).isNotEqualTo(emailTemplate2);
+    }
+}
```
Comment (pulkitkalra) [inline src/test/java/org/project36/qualopt/web/rest/EmailTemplateResourceIntTest.java]: Very good comprehensive tests 💯

Comment (victorlian) [reply src/test/java/org/project36/qualopt/web/rest/EmailTemplateResourceIntTest.java]: thanks but these were implemented by the original author, I cp and pasted here and changed the type basically. I'll take the credit :P

**Comment by victorlian** [comment]
@pulkitkalra Thanks for the feedback, does it mean the buttons would have to shift up as well? Or just the selector?
As for the same name issue, it is legal under the current implementation. I'll look into disabling that. shouldn't be too difficult

**Comment by pulkitkalra** [comment]
@victorlian I reckon the buttons should go up as well, but I'll leave that up to you to decide :)

**Comment by victorlian** [comment]
I'm a little worries about the screen real-estate... don't want to make the modal too big. Which is why I left the controls in the footer. 
Could you post a screenshot on where you want the buttons be (a rough position maybe?) Try not to extend the modal too much :)

**Comment by pulkitkalra** [comment]
I was thinking something like this (the back button can stay at the bottom). Having this at the top, it is clear which template you're modifying/ creating. What do you think?

image

**Comment by victorlian** [comment]
The one thing i don't like about this is the delete/update being far away from the save. (and so does the error message) Seems less coherent to me :(.
Shall we leave this for mike to decide ? :P
Should be an easy fix anyway

**Comment by pulkitkalra** [comment]
@victorlian oh whoops the save button, yeah that could go at the top as well? But this was just a suggestion, I'm happy to approve either way :)

**Comment by victorlian** [comment]
@pulkitkalra In that case I think I'll stick with the current design. It also needs to be consistent with the other "invitation email" tab, for which case the control is also at the bottom left corner. It doesn't make sense to shift that one to the top as well, because the "manage template" button should stay at the footer as it serves as a "forward" purpose, like the "next" button. 

For consistency reasons, I think I'm going to leave all buttons/form control to the bottoms.

**Comment by victorlian** [comment]
Thanks guys. 
It has been approved and is now ready to be merged. @softeng-701

**Comment by victorlian** [comment]
The commit above is the squashed version. Code can now be merged.

**Comment by softeng-701** [comment]
@victorlian Please resolve the conflicts!

**Comment by victorlian** [comment]
@softeng-701  merge conflict resolved, was a single line conflict, resolved by taking both changes, now able to be merged. 

(any updates on upgrading the size of the issue?)

**Comment by softeng-701** [comment]
@victorlian still the conflict exists. Can you kindly fix, thanks? 

I am not sure about which upgrade you are talking about, please explain, thanks.

**Comment by victorlian** [comment]
@softeng-701  I will fix the merge conflicts soon. Please read the first post and #21 for the upgrade request.

**Comment by softeng-701** [comment]
@victorlian Updated the size of #21 👍

**Pull request merged**